packages/x-components/src/x-modules/facets/components/filters/simple-filter.vue

```diff
@@ -56,48 +53,37 @@
    *
    * @public
    */
-  @Component({
+  export default defineComponent({
+    name: 'SimpleFilter',
+    xModule: facetsXModule.name,
     components: { RenderlessFilter },
-    mixins: [xComponentMixin(facetsXModule)]
-  })
-  export default class SimpleFilter extends Vue {
-    /** The filter data to render. */
-    @Prop({ required: true })
-    public filter!: SimpleFilterModel;
-
-    /**
-     * Additional events, with their payload, to emit when the filter is clicked.
-     *
-     * @public
-     */
-    @Prop()
-    public clickEvents?: Partial<XEventsTypes>;
-
-    /**
-     * The {@link XEventsTypes} to emit.
-     *
-     * @returns The events to emit when clicked.
-     * @internal
-     */
-    protected get _clickEvents(): Partial<XEventsTypes> {
-      return {
-        UserClickedASimpleFilter: this.filter,
-        ...this.clickEvents
-      };
-    }
+    props: {
+      /** The filter data to render. */
+      filter: {
+        type: Object as PropType<SimpleFilterModel>,
+        required: true
+      },
+      /** Additional events, with their payload, to emit when the filter is clicked. */
+      clickEvents: Object as PropType<Partial<XEventsTypes>>
+    },
+    setup(props) {
+      /** The {@link XEventsTypes} to emit. */
+      const clickEventsEmit = computed(() => ({
+        UserClickedASimpleFilter: props.filter,
+        ...props.clickEvents
+      }));
+
+      /** Dynamic CSS classes to apply to the component. */
+      const cssClasses = computed(() => ({
+        'x-simple-filter--is-selected': props.filter.selected
+      }));
```

Comment:
This could be done directly on the template.

Reply:
This option doesn't work either.